src/components/FeedbackForm.jsx

```diff
@@ -0,0 +1,185 @@
+      alert('Something went wrong! Please try again.');
+    }
+  };
+
+  const formSubmit = e => {
+    e.preventDefault();
+    setSubmitted(true);
+    const { firstName, lastName, mail, rating, feedback } = formValues;
+    if (firstName && lastName && mail && rating && feedback) {
+      setValidity(true);
+      setTimeout(function () {
+        localStorage.setItem('survey-form', JSON.stringify(formValues));
+        resetForm({ status: 200 });
+      }, 5000);
+    }
+  };
+
+  return (
+    <div>
+      {valid && <div className="success-msg">Form submitted successfully!</div>}
+      <form id="feedback-form" onSubmit={formSubmit}>
+        <fieldset>
+          <label htmlFor="firstName">First name:</label>
+          <input
+            type="text"
+            id="firstName"
+            name="firstName"
+            value={formValues.firstName}
+            onChange={changeVal}
+          />
+          {submitted && !formValues.firstName && (
+            <span className="error-msg">Name is empty or invalid!</span>
+          )}
+        </fieldset>
+        <fieldset>
+          <label htmlFor="lastName">Last name:</label>
+          <input
+            type="text"
+            id="lastName"
+            name="lastName"
+            value={formValues.lastName}
+            onChange={changeVal}
+          />
+          {submitted && !formValues.lastName && (
+            <span className="error-msg">Last name is empty or invalid!</span>
+          )}
+        </fieldset>
+        <fieldset>
+          <label htmlFor="mail">E-mail:</label>
+          <input type="email" id="mail" name="mail" value={formValues.mail} onChange={changeVal} />
+          {submitted && !formValues.mail && (
+            <span className="error-msg">E-mail is empty or invalid!</span>
+          )}
+        </fieldset>
+        <fieldset>
+          <label htmlFor="satisfactory-index">Rate our services</label>
+          <div className="radio-set-container">
+            <div className="radio-set">
+              <input
+                type="radio"
+                id="satisfactoryIndex1"
+                name="rating"
+                value="super-bad"
+                onChange={radioSelected}
+              />
+              <label htmlFor="satisfactoryIndex1">&#128169;</label>
+            </div>
+            <div className="radio-set">
+              <input
+                type="radio"
+                id="satisfactoryIndex2"
+                name="rating"
+                value="bad"
+                onChange={radioSelected}
+              />
+              <label htmlFor="satisfactoryIndex2">&#129314;</label>
+            </div>
+            <div className="radio-set">
+              <input
+                type="radio"
+                id="satisfactoryIndex3"
+                name="rating"
+                value="mediocre"
+                onChange={radioSelected}
+              />
+              <label htmlFor="satisfactoryIndex3">&#128547;</label>
```

Comment:
i would recommend to create config array for rating like this and then map inside jsx

```
const ratingConfig = [
  { value: 'super-bad', label: '&#128169;' }
  ....
];
...
ratingConfig.map(({ value, label }, index) => (
  <div className="radio-set">
  .......
```

Reply:
ok